### ISSUE TITLE
Remove github.com/mitchellh/mapstructure dependency

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#5858]
### Requirement

The project has been recently archived so we should remove any dependency from it.

### Problem

a

**Comment by gansheer** [comment]
There is a maintained fork for here that could be used : https://github.com/go-viper/mapstructure. But it is probably only a way to postpone the issue.

**Comment by squakez** [comment]
Yes, I finally opted to maintain the dependency. It's still used in a few deprecated blocks of codes, so, hopefully once we remove those deprecated blocks it will naturally disappear without any code refactoring.
